code/db_sync/src/wattrex_battery_cycler_db_sync/db_sync_node.py

```diff
@@ -77,12 +77,14 @@ def process_iteration(self) -> None:
         log.info("Processing iteration for experiment...") # pylint: disable=logging-fstring-interpolation
         self.fachade.push_experiments()
         self.fachade.push_gen_meas()
-        self.fachade.commit()
-        log.debug("Commit and push of gen and exp done")
-        self.fachade.push_ext_meas()
-        self.fachade.push_alarms()
-        self.fachade.push_status()
-        self.fachade.commit()
-        log.debug("Commit and push ext, alarms and status done")
-        self.fachade.delete_pushed_data()
-        self.fachade.update_last_connection()
+        try:
+            self.fachade.commit()
+            log.debug("Commit and push of gen and exp done")
+            self.fachade.push_ext_meas()
+            self.fachade.push_alarms()
+            self.fachade.push_status()
+            self.fachade.commit()
+            log.debug("Commit and push ext, alarms and status done")
+            self.fachade.delete_pushed_data()
+        except Exception as err:
```

Comment:
Do you really execute rollback if something fails? I think you should catch the exception and do the rollback in the  commit and delete_pushed_data functions